app/Http/Controllers/Admin/ModulesController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Contracts\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Artisan;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Str;
+use Laracasts\Flash\Flash;
+use Madnest\Madzipper\Madzipper;
+use App\Models\Module;
+
+class ModulesController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @param Request $request
+     *
+     * @return mixed
+     */
+    public function index()
+    {
+        $modules = Module::all();
+        return view('admin.modulesmanager.index', [
+            'modules' => $modules
+        ]);
+    }
+
+    public function addModule(Request $request)
+    {
+        if ($request->isMethod('get')) {
+            return view('admin.modulesmanager.add');
+        }
+        elseif ($request->isMethod('post')) {
+            $this->addModulePost($request);
+        }
```

Comment:
This `if`/`elseif` isn't the right way to do this. The routing needs to be configured so that `GET /admin/modules/create` routes to the `create()` method and `POST /admin/modules/create` routes to the `store()` method.

This is the convention to follow: https://laravel.com/docs/8.x/controllers#resource-controllers